src/scrabble.js

```diff
@@ -1,5 +1,231 @@
+    multiplier = 2
+  }
+
+  if (isTripleWord(givenWord)) {
+    multiplier = 3
+  }
+
+  if (isDoubleAndTripleWord(givenWord)) {
+    multiplier = 6
+  }
+
+  if (endLetterMultipliers(givenWord)) {
+    multiplier = 1
+  }
+
+  return multiplier
+}
+
+//  Check for double & triple letters
+//  Compares the i and i+2 characters to look for brackets with one character in between
+//  Adds 1 or 2 times the letter value if brackets are found
+
+function letterMultiplier(givenWord) {
+  let additionalLetterScore = 0
+  const lowerCaseString = givenWord.toLowerCase()
+  for (let i = 0; i < givenWord.length; i++) {
+    if (givenWord[i] === `{` && givenWord[i + 2] === `}`) {
+      additionalLetterScore += letterValues[lowerCaseString[i + 1]]
+    }
+  }
+  for (let i = 0; i < givenWord.length; i++) {
+    if (givenWord[i] === `[` && givenWord[i + 2] === `]`) {
+      additionalLetterScore += letterValues[lowerCaseString[i + 1]] * 2
+    }
+  }
+  return additionalLetterScore
+}
+
+//  Check for incomplete sets of brackets
+//  Counts number of open and close brackets
+//  If open does not equal close there must be an incomplete set of brackets
+
+function incompleteBrackets(givenWord) {
+  let curlyOpenCount = 0
+  let curlyClosedCount = 0
+  let squareOpenCount = 0
+  let squareClosedCount = 0
+  for (let i = 0; i < givenWord.length; i++) {
+    if (givenWord[i] === `{`) {
+      curlyOpenCount += 1
+    }
+    if (givenWord[i] === `}`) {
+      curlyClosedCount += 1
+    }
+    if (givenWord[i] === `[`) {
+      squareOpenCount += 1
+    }
+    if (givenWord[i] === `]`) {
+      squareClosedCount += 1
+    }
+  }
+  if (curlyOpenCount !== curlyClosedCount) {
+    return true
+  } else if (squareOpenCount !== squareClosedCount) {
+    return true
+  } else {
+    return false
+  }
+}
+
+//  check for any invalid characters
+//  [\] checks for square brackets
+//  {} checks for curly brackets
+//  \a-z checks for any alphanumeric characters in the a-z range
+//  ^ negates the set, returns true if any specified characters are found
+
+function invalidCharacterCheck(givenWord) {
+  const lowerCaseString = givenWord.toLowerCase()
+  return /[^[\]{}a-z]/g.test(lowerCaseString)
+}
+
+//  check if input fails for invalid characters or incorrect brackets
+
+function invalidGivenWord(givenWord) {
+  if (invalidCharacterCheck(givenWord) || incompleteBrackets(givenWord)) {
+    return true
+  } else {
+    return false
+  }
+}
+
+//  Adds together the base word score, any additional points from double/triple letters
+//  and multiplies everything by the word multiplier
+//  If any of the invalid input checks fail it exits at that point
+
+function scrabble(givenWord) {
+  let totalScore = 0
+  if (givenWord === null) {
+    return 0
+  }
+  totalScore = baseScore(givenWord)
+
+  totalScore += letterMultiplier(givenWord)
+
+  totalScore *= wordMultiplier(givenWord)
 
+  if (invalidGivenWord(givenWord)) {
+    return 0
+  } else {
+    return totalScore
+  }
 }
```

Comment:
Could have split up the functions into other files and used module.exports function to not have all your code in 1 long file (I am tired of scrolling bro...)